plugins/imgur.py

```diff
@@ -128,7 +128,11 @@ def export_submission(self,
             if is_album:
                 results['link_display'] = '[Imgur Album](https://imgur.com/a/%s)  \n' % album['id']
             else:
-                results['link_display'] = '[Imgur](%s)  \n' % images[0]['link'].replace('http', 'https')
+                picture_url = images[0]['link'].replace('http', 'https')
+                with urllib.request.urlopen(picture_url) as response:
```

Comment:
Use [`requests.head`](https://github.com/kupiakos/LapisMirror/blob/70676ac8745ff2a06970798356f0cb8b3b960251/plugins/puush.py#L73) to get the MIME type and compare against that.